plugins/catalog-backend/src/database/types.ts

```diff
@@ -36,6 +36,25 @@ export type DbEntityRequest = {
   relations: EntityRelationSpec[];
 };
 
+export type DbEntitiesRequest = {
+  filter?: EntityFilter;
+  pagination?: EntityPagination;
+};
+
+export type DbEntitiesResponse = {
+  entities: DbEntityResponse[];
+  pageInfo: DbPageInfo;
+};
+
+export type DbPageInfo =
+  | {
+      hasNext: false;
+    }
+  | {
+      hasNext: true;
+      endCursor: string;
```

Comment:
Any thoughts on using `nextCursor?: string` instead?

Reply:
I googled around a bit and seems that `hasNext[Page]` and `endCursor` were what graphql and the likes were using. And there's a logic to having a cursor to the `end` and then saying in the next request that I only wants what comes `after` that `end`.

Reply:
Well was thinking that nextCursor would be the end one, but could see why that's a bit awkward.

But alright, should we stick to https://graphql.org/learn/pagination/ as standard for pagination perhaps then? Switching `edges` to `items` for non-graph data?

Reply:
So that we look at it as us returning a Connection? Sure, we could name them `items`, but we could have a `?onlyIds=true` or a different "sister" endpoint that returns `edges` and can be more heavily optimized.

Reply:
We could give backend-common a `PaginatedResponse` type with `items` and `pageInfo`, and a `PageInfo` type, to help with formulating standard responses

Reply:
Sure, but was also thinking we align the naming, so hasNextPage instead

Reply:
That makes sense.  I could fix that right away actually

Reply:
https://github.com/backstage/backstage/pull/5021

Reply:
Zalando's RESTful API guidelines have a proposal for cursors and pagination links (https://opensource.zalando.com/restful-api-guidelines/#pagination).